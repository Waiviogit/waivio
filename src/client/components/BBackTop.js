import React from 'react';
import PropTypes from 'prop-types';
import { BackTop, Button } from 'antd';
import classNames from 'classnames';
import './Chat/Chat';
import './BBackTop.less';

export default function BBackTop({
  className,
  isModal,
  isChat,
  authentication,
  openChat,
  messagesCount,
  ...otherProps
}) {
  return (
    <div className="BBackTop">
      <div
        className={classNames(className, 'BBackTop__container', {
          'BBackTop__container--shifted': isModal,
        })}
      >
        {authentication ? (
          <div className={classNames('BBackTop__chat-button')}>
            <Button
              onClick={openChat}
              type="primary"
              shape="circle"
              icon={
                // eslint-disable-next-line no-nested-ternary
                !isChat ? (!messagesCount ? 'message' : null) : 'close'
              }
            >
              {// eslint-disable-next-line no-nested-ternary
              !isChat && messagesCount ? (messagesCount > 99 ? '99+' : messagesCount) : null}
            </Button>
          </div>
        ) : null}
        <BackTop className="BBackTop_button" {...otherProps}>
          <i className="iconfont icon-back-top" />
        </BackTop>
      </div>
    </div>
  );
}

BBackTop.propTypes = {
  className: PropTypes.string,
  isModal: PropTypes.bool,
  isChat: PropTypes.bool.isRequired,
  openChat: PropTypes.func.isRequired,
  authentication: PropTypes.bool.isRequired,
  messagesCount: PropTypes.number,
};

BBackTop.defaultProps = {
  className: '',
  isModal: false,
  messagesCount: 0,
};
