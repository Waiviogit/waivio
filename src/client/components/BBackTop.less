@import (reference) '../styles/custom.less';
@import (reference) '../styles/modules/variables.less';

.BBackTop {
  position: fixed;
  width: 100%;
  height: 0;
  bottom: 60px;
  right: 20px;
  z-index: @backtop-z-index;

  .BBackTop__container {
    position: relative;
    display: flex;
    justify-content: flex-end;
    margin: 0 auto;
    max-width: @container-width;
    @media @large {
      padding: 0;
    }

    &--shifted {
      max-width: 935px;
    }
  }

  &_button {
    position: static !important;
    float: right;

    i {
      color: @primary-color !important;
      font-size: 40px;
      line-height: 40px;
      background-color: white;
      border-radius: 50%;
      height: 37px;
      display: block;
      width: 37px;

      &:hover {
        color: @blue-navyBlue !important;
      }
    }
  }
  &__chat-button {
    margin-right: 5px;
  }
}
.BBackTop__chat-button > .ant-btn-circle {
  width: 40px;
  height: 40px;
}
