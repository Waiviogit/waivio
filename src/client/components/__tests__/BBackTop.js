import React from 'react';
import { act } from 'react-dom/test-utils';
import { shallow } from 'enzyme';
import BBackTop from '../BBackTop';

describe('<BBackTop />', () => {
  it('renders without exploding', () => {
    const props = {
      className: 'comments',
      isModal: 42,
    };
    const wrapper = shallow(<BBackTop {...props} />);
    act(() => {
      wrapper.update();
    });
    expect(wrapper).toMatchSnapshot();
  });
});
