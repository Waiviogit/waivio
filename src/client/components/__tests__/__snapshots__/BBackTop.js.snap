// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<BBackTop /> renders without exploding 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <BBackTop
    className="comments"
    isModal={false}
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "checkPropTypes": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": <div
        className="comments BBackTop__container"
      >
        <BackTop
          className="BBackTop_button"
          visibilityHeight={400}
        >
          <i
            className="iconfont icon-back-top"
          />
        </BackTop>
      </div>,
      "className": "BBackTop",
    },
    "ref": null,
    "rendered": Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": <BackTop
          className="BBackTop_button"
          visibilityHeight={400}
        >
          <i
            className="iconfont icon-back-top"
          />
        </BackTop>,
        "className": "comments BBackTop__container",
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "children": <i
            className="iconfont icon-back-top"
          />,
          "className": "BBackTop_button",
          "visibilityHeight": 400,
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "className": "iconfont icon-back-top",
          },
          "ref": null,
          "rendered": null,
          "type": "i",
        },
        "type": [Function],
      },
      "type": "div",
    },
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": <div
          className="comments BBackTop__container"
        >
          <BackTop
            className="BBackTop_button"
            visibilityHeight={400}
          >
            <i
              className="iconfont icon-back-top"
            />
          </BackTop>
        </div>,
        "className": "BBackTop",
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": <BackTop
            className="BBackTop_button"
            visibilityHeight={400}
          >
            <i
              className="iconfont icon-back-top"
            />
          </BackTop>,
          "className": "comments BBackTop__container",
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "children": <i
              className="iconfont icon-back-top"
            />,
            "className": "BBackTop_button",
            "visibilityHeight": 400,
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "className": "iconfont icon-back-top",
            },
            "ref": null,
            "rendered": null,
            "type": "i",
          },
          "type": [Function],
        },
        "type": "div",
      },
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "legacyContextMode": "parent",
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getChildContext": Object {
            "calledByRenderer": false,
          },
          "getDerivedStateFromError": true,
          "getDerivedStateFromProps": Object {
            "hasShouldComponentUpdateBug": false,
          },
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
    Symbol(enzyme.__providerValues__): undefined,
  },
  Symbol(enzyme.__providerValues__): Map {},
}
`;
